docs/framework/docker/console.md

```diff
@@ -0,0 +1,139 @@
+
+You need to be familiar with some Docker terms before you begin working
+on moving your application to a container.
+
+> A **Docker image** is a read-only template that defines the environment
+> for a running container, including the OS, system components, and application(s).
+
+One important feature of Docker images is that images are composed from a
+base image. Each new image adds a small set of features to an existing
+image. 
+
+> A **Docker container** is a running instance of an image. 
+
+You scale an application by running the same image in many containers.
+Conceptually, this is similar to running the same application in multiple
+hosts.
+
+You can learn more about the Docker architecture by reading the 
+[Docker Overview](https://docs.docker.com/engine/understanding-docker/)
+on the Docker site. 
+
+## Getting started
+You'll need to have Docker for Windows, version 1.12 Beta 26 or higher to support Windows containers. By default, Docker enables Linux based containers; switch to Windows containers by right clicking the Docker icon in the system tray and select "Switch to Windows Containers...". Docker will run the process to change and a restart may be required.
+
+Moving your console application is a matter of a few steps.
+
+1. Build the application
+1. Creating a Dockerfile for the image
+1. Process to build and run the Docker container
+
+## Building the application
+Typically console applications are distributed through an installer, FTP, or File Share deployment. Here the assets need to be compiled and staged to a location that can be used when the Docker image is created.
+
+In **build.ps1** the script uses MSBuild to compile the application with a configuration of *Release* and publishes the assets to the *publish* folder.
+
+```
+function Invoke-MSBuild ([string]$MSBuildPath, [string]$MSBuildParameters) {
+    Invoke-Expression "$MSBuildPath $MSBuildParameters"
+}
+
+Invoke-MSBuild -MSBuildPath "MSBuild.exe" -MSBuildParameters ".\ConsoleRandomAnswerGenerator.csproj /p:OutputPath=.\publish /p:Configuration=Release"
+```
+
+## Creating the Dockerfile
+The base image used for a Console .NET Framework applications is `microsoft/windowsservercore`, publically available on [Docker Hub](https://hub.docker.com/r/microsoft/windowsservercore/). The image is the minimal installation of Windows Server 2016 and serves as the base OS image for Windows containers.
+
+```
+FROM microsoft/windowsservercore
+ADD publish/ /
+ENTRYPOINT ConsoleRandomAnswerGenerator.exe
+```
+
+The `ADD` in the file copies the application assets from the **publish** folder to root folder of the container and last; setting the `ENTRYPOINT` of the image states that this is the command or application that will run when the container starts. 
+
+## Creating the image
+Adding the following code to the **build.ps1** script, when run the Docker image called `console-random-answer-generator` is created after the compilation is complete.
```

Comment:
> Adding the following code to the **build.ps1** script, when run the Docker image called console-random-answer-generator is created after the compilation is complete.

The sentence took me a few reads to understand what was being articulated. Perhaps:

Adding the following code to the **build.ps1** script will create a Docker image called `console-random-answer-generator` once the compilation is complete.

Then, add a second step telling them to run the file.